pennylane/devices/default_qubit_torch.py

```diff
@@ -0,0 +1,329 @@
+            If ``shots > 0`` is used, the ``diff_method="backprop"``
+            QNode differentiation method is not supported and it is recommended to consider
+            switching device to ``default.qubit`` and using ``diff_method="parameter-shift"``.
+        torch_device: the device on which the computation will be run, 'cpu' or 'cuda'
+            default='cpu'
+    """
+
+    name = "Default qubit (Torch) PennyLane plugin"
+    short_name = "default.qubit.torch"
+
+    parametric_ops = {
+        "PhaseShift": torch_ops.PhaseShift,
+        "ControlledPhaseShift": torch_ops.ControlledPhaseShift,
+        "RX": torch_ops.RX,
+        "RY": torch_ops.RY,
+        "RZ": torch_ops.RZ,
+        "MultiRZ": torch_ops.MultiRZ,
+        "Rot": torch_ops.Rot,
+        "CRX": torch_ops.CRX,
+        "CRY": torch_ops.CRY,
+        "CRZ": torch_ops.CRZ,
+        "SingleExcitation": torch_ops.SingleExcitation,
+        "SingleExcitationPlus": torch_ops.SingleExcitationPlus,
+        "SingleExcitationMinus": torch_ops.SingleExcitationMinus,
+        "DoubleExcitation": torch_ops.DoubleExcitation,
+        "DoubleExcitationPlus": torch_ops.DoubleExcitationPlus,
+        "DoubleExcitationMinus": torch_ops.DoubleExcitationMinus,
+    }
+
+    C_DTYPE = torch.complex128
+    R_DTYPE = torch.float64
+
+    _abs = staticmethod(torch.abs)
+    _einsum = staticmethod(torch.einsum)
+    _flatten = staticmethod(torch.flatten)
+    _reshape = staticmethod(torch.reshape)
+    _roll = staticmethod(torch.roll)
+    _stack = staticmethod(lambda arrs, axis=0, out=None: torch.stack(arrs, axis=axis, out=out))
+    _tensordot = staticmethod(
+        lambda a, b, axes: torch.tensordot(
+            a, b, axes if isinstance(axes, int) else tuple(map(list, axes))
+        )
+    )
+    _transpose = staticmethod(lambda a, axes=None: a.permute(*axes))
+    _asnumpy = staticmethod(lambda x: x.cpu().numpy())
+    _conj = staticmethod(torch.conj)
+    _imag = staticmethod(torch.imag)
+    _norm = staticmethod(torch.norm)
+    _flatten = staticmethod(torch.flatten)
+
+    def __init__(self, wires, *, shots=None, analytic=None, torch_device="cpu"):
+        self._torch_device = torch_device
+        super().__init__(wires, shots=shots, cache=0, analytic=analytic)
+
+        # Move state to torch device (e.g. CPU, GPU, XLA, ...)
+        self._state.requires_grad = True
+        self._state = self._state.to(self._torch_device)
+        self._pre_rotated_state = self._state
+
+    @staticmethod
+    def _asarray(a, dtype=None):
+        if isinstance(a, list):
+            # here be dragons... but it seem to give the results we want
+            if not isinstance(a[0], torch.Tensor):
+                res = np.asarray(a)
+                res = torch.from_numpy(res)
+            else:
+                res = torch.cat([torch.reshape(i, (-1,)) for i in a], dim=0)
+            res = torch.cat([torch.reshape(i, (-1,)) for i in res], dim=0)
```

Comment:
Hi @PCesteban, just wanted to make sure that the suggested updates in
* https://github.com/PennyLaneAI/pennylane/pull/1360/commits/91102689a8d5dbe681dd59be8c27c423e35d3f51
* https://github.com/PennyLaneAI/pennylane/pull/1360/commits/1185952c0135a835cb7e3ed028149519a0d935ea

aligned well with the intentions for this method. We've noticed that previously the device might output `numpy` arrays, instead of `torch` tensors (hence the suggestions from before). Let us know if we've missed a use case here.

Reply:
Hi @antalszava, I don't think there is any missed use case to the best of my knowledge. `torch` tensors are a better fit for this method.

FYI   
@Slimane33 @arshpreetsingh